### ISSUE TITLE
Weird Empty Bomb thing

### ISSUE THREAD

**Issue by r0bd0g** [TestRunnerSRL/OoT-Randomizer#1448]
https://clips.twitch.tv/VivaciousPlumpGarlicTinyFace-11MHZYWzvaMClQbp

For full context, this entire restream will probably be on youtube eventually.

**Comment by r0bd0g** [comment]
After watching the full vod on youtube a little closer, I think this actually started prior to warping to crater, when I went to bomb open the colossus fairy entrance.

**Comment by fenhl** [comment]
What's the actual issue here? I don't see anything out of the ordinary in the clip.

**Comment by r0bd0g** [comment]
The bombs are placing/throwing immediately. This is a known thing that can happen with the empty bomb glitch. We have a fix for it but I think it must not be perfect? This was restreamed on ZSR, I forget which race, but it should be archived on YouTube somewhere. I watched it back a while ago and it actually may have started happening a bit before this clip starts.

**Comment by djevangelia** [comment]
I looked at it but I think this was a controller issue or similar. Looking at the item use process:

**Normal use bomb (held item is not already bomb)**
Frame 1: Set held item.
Frame 2: Set item actions.
Frame 3: Set upper action to CarryActor, which calls throw/drop action handler. _This is first possible frame to throw/drop the bomb after using it._

**Bomb removal**
If the bomb is thrown/dropped, or explodes in Link's hands _when not running_, the held item and item actions are set to none.
In 1.0, if bomb explodes in Link's hands when running (never was thrown/dropped), these things do not happen. (This is because taking damage while running doesn't call Player_ResetStatesHeldActor, see func_80837C0C.)
Changing scene will set item actions and held item to none as part of Player_Init.

**Instant bomb**
In 1.0 bug, bomb will be used with bomb already as held item and upper action still CarryActor.
CarryActor calls throw/drop action handler already on frame 1 of item change, and the item button was pressed this frame - instant throw/drop.

**Bug fix**
The randomizer fix for this is to make the bomb set held item and item actions to none when deleting itself, if player is still holding the bomb. It doesn't introduce any new possibility of setting bomb as item action or held item.

Looking at recording:
- It's not possible for an item bug like quickdraw to carry over between scenes, so if it started before DMC, ...
- Any oddity before breaking the rock and entering the grotto would be removed from using the sword inside and enter/exit
- First use of bomb in a scene cannot quickdraw, because bomb must have been held item for player to be in such a state
- The instant throw/drop will set item action and held item to none, so it cannot occur two times in a row

In end of this play I recorded both 1.0 bug (frame 1 throw) and first possible frame throw (frame 3) but quality aside I can't see any difference. There's a sound difference but not available in the bug report recording. https://www.youtube.com/watch?v=6-OmJg4RUOE

**Comment by r0bd0g** [comment]
Yeah I have no idea what this was. I haven't changed my setup or my controller. It never happened again. I could imagine myself accidentally double-tapping the c button maybe once, but the number of times it happened in a row here it doesn't seem likely to me that I messed up that many times in a row idk.

**Comment by fenhl** [comment]
Given the analysis, the fact that it only happened once, and the lack of input display, I don't think this is something we can realistically fix even if it wasn't a setup issue, so I'm going to close this pending more data being provided.
